src/iterator.cr

```diff
@@ -639,6 +727,17 @@ module Iterator(T)
     end
   end
 
+  # Returns an iterator that takes puts together the corresponding elements
```

Comment:
So for the style 2 version I'd go with "Returns an iterator that returns the elements of this iterator and the passed one pairwise."